lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2128,23 +2110,21 @@ namespace Js
         wchar_t *outStr = builder.DangerousGetWritableBuffer();
 
         wchar_t* outStrLim = outStr + count;
+        wchar_t *o = outStr;
+
+        while (o < outStrLim)
+        {
+            *o++ = *inStr++;
+        }
 
-        //TODO the mapping based on unicode tables needs review for ES5.
-        // mapping is not always 1.1
         if(toCase == ToUpper)
         {
-            while (outStr < outStrLim)
-            {
-                *outStr++ = CharToUpperCase(*inStr++);
-            }
+            CharUpperBuffW(outStr, count);
```

Comment:
Test results for CharLowerW(), CharLowerBuffW(), CharUpperW(), CharUpperBuffW():

[u.txt](https://github.com/Microsoft/ChakraCore/files/116575/u.txt)